rhino/src/main/java/org/mozilla/javascript/Interpreter.java

```diff
@@ -2273,6 +2352,28 @@ private static Object interpretLoop(Context cx, CallFrame frame, Object throwabl
                             case Token.THIS:
                                 stack[++stackTop] = frame.thisObj;
                                 continue Loop;
+                            case Token.SUPER:
+                                {
+                                    // See 9.1.1.3.5 GetSuperBase
+
+                                    // If we are referring to "super", then we always have an
+                                    // activation
+                                    // (this is done in IrFactory). The home object is stored as
+                                    // part of the
+                                    // activation frame to propagate it correctly for nested
+                                    // functions.
+                                    Scriptable homeObject = getCurrentFrameHomeObject(frame);
+                                    if (homeObject == null) {
+                                        // This if is specified in the spec, but I cannot imagine
+                                        // how the home object will ever be null since `super` is
+                                        // legal _only_ in method definitions, where we do have a
+                                        // home object!
+                                        stack[++stackTop] = Undefined.instance;
```

Comment:
We agree. But, as discussed by comment, I cannot imagine how that line would ever be hit - the `super` token should only be allowed inside methods, which will always have a home object set, and all the activations stuff 
should work correctly and make the home object accessible. Otherwise, we get syntax errors before we reach the interpreter.

So, this is a case of where we implemented the spec word by word, but we can't imagine that this will ever happen given the actual implementation of the interpreter.